_pytest/python.py

```diff
@@ -235,8 +236,7 @@ def getmodpath(self, stopatmodule=True, includemodule=False):
                 continue
             name = node.name
             if isinstance(node, Module):
-                assert name.endswith(".py")
-                name = name[:-3]
+                name = os.path.splitext(name)[0]
```

Comment:
About a test for this:

I was considering writing a unittest that basically creates a `Module` instance and calls `getmodpath`.

But after thinking some more about it I think we need an integration test that demonstrates it is possible to collect a file without a `.py` extension as a `Module` object and the entire machinery works, otherwise we might end up introducing a regression later in some other part of the code and breaking this use case without realizing it.

I suggest to create a test that has a `conftest.py` file which implements the `pytest_collect_file` hook like this (untested):

```python
def test_collects_different_extensions_as_python(testdir):
    """Ensure we can collect files with different extensions as .py files (#2369)""".
    testdir.makeconftest("""
        def pytest_collect_file(path, parent):
            ext = path.ext
            if ext == ".hy":
                return Module(path, parent)
    """)
    testdir.writefile("""
        # ... 1 test here in Hy ...
    """, ext=".hy")
    result = testdir.runpytest()
    result.stdout.fnmatch_lines('*1 passed*')
```